client/components/AddTestPlanToQueue/AddTestPlanToQueueModal.jsx

```diff
@@ -0,0 +1,273 @@
+import React, { useState, useEffect } from 'react';
+import { Button, Modal, Form } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { useQuery, useMutation } from '@apollo/client';
+import {
+    ADD_TEST_QUEUE_MUTATION,
+    POPULATE_ADD_TEST_PLAN_TO_QUEUE_MODAL_QUERY
+} from '../TestQueue/queries';
```

Comment:
> When they're all extracted out it become harder to track! And the queries file might get uncomfortably long and hard to organize. For that reason I think it might be better to keep the queries in the component file.

Well a counter-point here is that having all the GQL statements in the component may potentially make it unnecessarily long and difficult to navigate.

The overall idea here was that the parent folder for that very parent and/or its children would have an associated query file of all the queries related to them. This may not be obvious due to the folder structure but `/components/AddTestPlanToQueue` would be a direct child of `/components/TestQueue` which is why I referenced the GQL statements from the `/component/TestQueue/queries` file in this instance.

> On the other hand, putting them in one file would make things like GraphQL fragments easier to pass around between queries. Maybe there are other good reasons I haven't thought of. What do you think?

I'm certainly a fan of having a dedicated file as a 'separation of concerns' for related queries but just for my clarification, did you mean a single file across the entire application?

Reply:
> The overall idea here was that the parent folder for that very parent and/or its children would have an associated query file of all the queries related to them.

Ah, interesting - you're right, I missed the implication there.

Okay, that definitely mitigates the point about the query file getting too long, then. There is a point where it won't need to be extended any more.

I thought of another pro to this approach, which is that it might potentially be easier to hook up to suspense for data fetching if that ever makes it into React:

The problem with suspense for data fetching is that it doesn't solve the "waterfall" problem, where a parent component makes a request, then loads a child component which makes a request, then loads another child component which makes a request. The requests happen sequentially instead of in parallel. So to reap the benefits of suspense, you have to group the queries for child components in to a parent component ... which is quite similar to what we have here.

All that to say, I think I get why you went with it - I'm on board!

Reply:
> The problem with suspense for data fetching is that it doesn't solve the "waterfall" problem, where a parent component makes a request, then loads a child component which makes a request, then loads another child component which makes a request. The requests happen sequentially instead of in parallel. So to reap the benefits of suspense, you have to group the queries for child components in to a parent component ... which is quite similar to what we have here.

Yep, I agree that it could help out there. Good observation